kinescope/kinescope.py

```diff
@@ -0,0 +1,110 @@
+"""This XBlock embeds content from Kinescope through Iframes"""
+
+import pkg_resources
+from web_fragments.fragment import Fragment
+from xblock.core import XBlock
+from xblock.fields import Integer, Scope, String
+try:
+    from xblock.utils.studio_editable import StudioEditableXBlockMixin
+except ModuleNotFoundError: # For compatibility with Palm and earlier
+    from xblockutils.studio_editable import StudioEditableXBlockMixin
+from xblock.validation import ValidationMessage
+
+
+from .utils import _, is_url, parse_valid_kinescope_url
+
+
+class KinescopeXBlock(StudioEditableXBlockMixin, XBlock):
+    """
+    This XBlock renders Iframe for Kinescope videos.
+    """
+
+    display_name = String(
+        display_name=_("Display name"),
+        default=_("Kinescope"),
+        scope=Scope.settings,
+        help=_("Display name for this XBlock."),
+    )
+
+    video_link = String(
+        display_name="Video Link/URL",
+        default="",
+        scope=Scope.content,
+        help=_(
+            "Video link copied from Kinescope dashboard. The video id below is extracted "
+            "from this link if provided"
+        )
+    )
+
+    video_id = String(
+        display_name=_("Video ID"),
+        default="",
+        scope=Scope.content,
+        help=_(
+            "UUID of the video to embed."
+        ),
+        resettable_editor=False
+    )
+
+    editable_fields = ('display_name', 'video_link', 'video_id')
+
+
+    def clean_studio_edits(self, data):
+        """
+        Parse video_link if provyided and populate video_id
+        """
+        if "video_link" in data and (video_id := parse_valid_kinescope_url(data["video_link"])):
+            data["video_id"] = video_id
```

Comment:
I agree with you, that this behaviour can be quite confusing.

I have therefore removed the `video_id` field completely and now only accept video links. The reason for this is:
1. Kinescope has remove the option of copying just the `Video ID/UUID` from their dashboard UI and now only supports copying video link. Also, their [wordpress plugin doc](https://en-gb.wordpress.org/plugins/kinescope/#description) also only supports adding a video url.
2. Accepting both video id and video link in the same field will make it difficult to validate the url.

Reply:
@kaustavb12,
> Accepting both video id and video link in the same field will make it difficult to validate the url.

Not really. If the whole value contains only ID-friendly characters, we can treat it as an ID. Otherwise, we move it to the URL validation.

Reply:
That's a good point. Though I'll prefer to keep it this way for now, since there is no way to get just the video id directly from the Kinescope dashboard. I could point this out to the client and make changes based on their feedback.